robots/bebop.h

```diff
@@ -0,0 +1,1188 @@
+            }
+        }
+
+        m_IsConnected = false;
+    }
+}
+
+/*
+ * Start controlling this drone with a joystick.
+ */
+void
+Bebop::addJoystick(HID::Joystick &joystick)
+{
+    joystick.addHandler([this](const HID::JAxis axis, const float value) {
+        return onAxisEvent(axis, value);
+    });
+    joystick.addHandler([this](const HID::JButton button, const bool pressed) {
+        return onButtonEvent(button, pressed);
+    });
+}
+
+/*
+ * Send take-off command.
+ */
+void
+Bebop::takeOff()
+{
+    if (m_IsConnected) {
+        std::cout << "Taking off..." << std::endl;
+        DRONE_COMMAND_NO_ARG(sendPilotingTakeOff);
+
+        if (m_FlightEventHandler) {
+            m_FlightEventHandler(true);
+        }
+    }
+}
+
+/*
+ * Send land command.
+ */
+void
+Bebop::land()
+{
+    if (m_IsConnected) {
+        std::cout << "Landing..." << std::endl;
+        DRONE_COMMAND_NO_ARG(sendPilotingLanding);
+
+        if (m_FlightEventHandler) {
+            m_FlightEventHandler(false);
+        }
+    }
+}
+
+Bebop::VideoStream &
+Bebop::getVideoStream()
+{
+    startStreaming();
+    return *m_VideoStream;
+}
+
+inline degree_t
+Bebop::getMaximumTilt()
+{
+    if (!m_IsConnected) {
+        throw std::runtime_error("Not connected to drone");
+    }
+    return m_TiltLimits.getCurrent();
+}
+
+inline Limits<degree_t> &
+Bebop::getTiltLimits()
+{
+    if (!m_IsConnected) {
+        throw std::runtime_error("Not connected to drone");
+    }
+    return m_TiltLimits.getLimits();
+}
+
+inline meters_per_second_t
+Bebop::getMaximumVerticalSpeed()
+{
+    if (!m_IsConnected) {
+        throw std::runtime_error("Not connected to drone");
+    }
+    return m_VerticalSpeedLimits.getCurrent();
+}
+
+inline Limits<meters_per_second_t> &
+Bebop::getVerticalSpeedLimits()
+{
+    if (!m_IsConnected) {
+        throw std::runtime_error("Not connected to drone");
+    }
+    return m_VerticalSpeedLimits.getLimits();
+}
+
+inline degrees_per_second_t
+Bebop::getMaximumYawSpeed()
+{
+    if (!m_IsConnected) {
+        throw std::runtime_error("Not connected to drone");
+    }
+    return m_YawSpeedLimits.getCurrent();
+}
+
+inline Limits<degrees_per_second_t> &
+Bebop::getYawSpeedLimits()
+{
+    if (!m_IsConnected) {
+        throw std::runtime_error("Not connected to drone");
+    }
+    return m_YawSpeedLimits.getLimits();
+}
+
+inline void
+Bebop::setMaximumTilt(const degree_t newValue)
+{
+    DRONE_COMMAND(sendPilotingSettingsMaxTilt, newValue.value());
+}
+
+inline void
+Bebop::setMaximumVerticalSpeed(const meters_per_second_t newValue)
+{
+    DRONE_COMMAND(sendSpeedSettingsMaxVerticalSpeed, newValue.value());
+}
+
+inline void
+Bebop::setMaximumYawSpeed(const degrees_per_second_t newValue)
+{
+    DRONE_COMMAND(sendSpeedSettingsMaxRotationSpeed, newValue.value());
+}
+
+/*
+ * Set drone's pitch, for moving forwards and backwards.
+ */
+void
+Bebop::setPitch(const float pitch)
+{
+    assert(pitch >= -1.0f && pitch <= 1.0f);
```

Comment:
With floating point maths being what it is, I think it would be a bit friendlier to do
```c++
pitch = std::max(-1.0f, std::min(pitch, 1.0f));
```
....Which is already a good case for not passing value parameters as const...